
<!DOCTYPE html>
<!--[if IEMobile 7 ]><html class="no-js iem7"><![endif]-->
<!--[if lt IE 9]><html class="no-js lte-ie8"><![endif]-->
<!--[if (gt IE 8)|(gt IEMobile 7)|!(IEMobile)|!(IE)]><!--><html class="no-js" lang="en"><!--<![endif]-->
<head>
  <meta charset="utf-8">
  <title>Introducción al análisis de microarreglos - The Grumpy Biochemist</title>
  <meta name="author" content="Felipe Antonio Riveroll Aguirre">

  
  <meta name="description" content="Los datos que utilizaremos vienen de un estudio de Chiaretti et al. sobre la leucemia linfoblastica aguda (ALL), la cual fue conducida con chips de &hellip;">
  

  <!-- http://t.co/dKP3o1e -->
  <meta name="HandheldFriendly" content="True">
  <meta name="MobileOptimized" content="320">
  <meta name="viewport" content="width=device-width, initial-scale=1">

  
  <link rel="canonical" href="http://friveroll.github.io/blog/2012/05/20/introduccion-al-analisis-de-microarreglos/">
  <link href="/favicon.png" rel="icon">
  <link href="/stylesheets/screen.css" media="screen, projection" rel="stylesheet" type="text/css">
  <script src="/javascripts/modernizr-2.0.js"></script>
  <script src="/javascripts/ender.js"></script>
  <script src="/javascripts/octopress.js" type="text/javascript"></script>
  <link href="/atom.xml" rel="alternate" title="The Grumpy Biochemist" type="application/atom+xml">
  <!--Fonts from Google"s Web font directory at http://google.com/webfonts -->
<link href="http://fonts.googleapis.com/css?family=PT+Serif:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<link href="http://fonts.googleapis.com/css?family=PT+Sans:regular,italic,bold,bolditalic" rel="stylesheet" type="text/css">
<!-- mathjax config similar to math.stackexchange -->
<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  jax: ["input/TeX", "output/HTML-CSS"],
  tex2jax: {
    inlineMath: [ ['$', '$'] ],
    displayMath: [ ['$$', '$$']],
    processEscapes: true,
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre', 'code']
  },
  messageStyle: "none",
  "HTML-CSS": { preferredFont: "TeX", availableFonts: ["STIX","TeX"] }
});
</script>
<script src="http://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS_HTML" type="text/javascript"></script>
  
  <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-79218613-1']);
    _gaq.push(['_trackPageview']);

    (function() {
      var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
      ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
  </script>


</head>

<body   >
  <header role="banner"><hgroup>
  <h1><a href="/">The Grumpy Biochemist</a></h1>
  
    <h2>A blog by Felipe Riveroll Aguirre</h2>
  
</hgroup>

</header>
  <nav role="navigation"><ul class="subscription" data-subscription="rss">
  <li><a href="/atom.xml" rel="subscribe-rss" title="subscribe via RSS">RSS</a></li>
  
</ul>
  
<form action="http://google.com/search" method="get">
  <fieldset role="search">
    <input type="hidden" name="q" value="site:friveroll.github.io" />
    <input class="search" type="text" name="q" results="0" placeholder="Search"/>
  </fieldset>
</form>
  
<ul class="main-navigation">
  <li><a href="/">Blog</a></li>
  <li><a href="/blog/archives">Archives</a></li>
  <li><a href="/pdfs/CV.pdf">CV</a></li>
  <li><a href="http://feliperiveroll.info">About</a></li>
</ul>

</nav>
  <div id="main">
    <div id="content">
      <div>
<article class="hentry" role="article">
  
  <header>
    
      <h1 class="entry-title">Introducción Al Análisis De Microarreglos</h1>
    
    
      <p class="meta">
        








  


<time datetime="2012-05-20T21:16:00-05:00" pubdate data-updated="true">May 20<span>th</span>, 2012</time>
        
         | <a href="#disqus_thread">Comments</a>
        
      </p>
    
  </header>


<div class="entry-content"><p>Los datos que utilizaremos vienen de un estudio de Chiaretti et al.  sobre la leucemia linfoblastica aguda (ALL), la cual fue conducida con chips de microarreglos HG-U95Av2 de Aymetrix. El paquete de datos ALL contiene los datos de expresion del experimento, normalizados con rma (las intensidades estan en escala log2), junto con las anotaciones de las muestras. </p>

<p>Para ello utilizaremos R /Bioconductor Si no has instalado Bioconductor previemente necesitas correr las siguientes líneas</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">source<span class="p">(</span><span class="s">&quot;http://bioconductor.org/biocLite.R&quot;</span><span class="p">)</span>
</span><span class="line">biocLite<span class="p">()</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>
<p>Necesitaremos instalar algunas librerías de Bioconductor</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">biocLite<span class="p">(</span>c<span class="p">(</span><span class="s">&quot;affy&quot;</span><span class="p">,</span><span class="s">&quot;ALL&quot;</span><span class="p">,</span><span class="s">&quot;annotate&quot;</span><span class="p">,</span><span class="s">&quot;hgu95av2.db&quot;</span><span class="p">,</span><span class="s">&quot;multtest&quot;</span><span class="p">,</span><span class="s">&quot;topGO&quot;</span><span class="p">,</span><span class="s">&quot;genefilter&quot;</span><span class="p">))</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>
<p>Ahora, cargamos los datos de microarreglos y los asociamos a un set de expresión con</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">library<span class="p">(</span>ALL<span class="p">)</span>
</span><span class="line">data<span class="p">(</span>ALL<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>
<p>Y para conocer la dimensión de la matriz de expresión</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">dim<span class="p">(</span>exprs<span class="p">(</span>ALL<span class="p">))</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><code>[1] 12625 128</code></p>

<p>Por tanto, se trata de 128 columnas (muestras) con 12,625 filas (sondas) Nos interesa seleccionar todas las sondas de las células B (que pueden ser identicadas por la columna BT del ExpressionSet ALL).</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">table<span class="p">(</span>ALL<span class="o">$</span>BT<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">B B1 B2 B3 B4  <span class="k-Variable">T</span> T1 T2 T3 T4
</span><span class="line"><span class="m">5</span> <span class="m">19</span> <span class="m">36</span> <span class="m">23</span> <span class="m">12</span>  <span class="m">5</span>  <span class="m">1</span> <span class="m">15</span> <span class="m">10</span>  <span class="m">2</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Ademas nos interesa la comparacion de las sondas con la fusión del gen BCR/ABL resultante de la traslocacion de los cromosomas 9 y 22</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">table<span class="p">(</span>ALL<span class="o">$</span>mol.biol<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>
<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">ALL1<span class="o">/</span>AF4  BCR<span class="o">/</span>ABL E2A<span class="o">/</span>PBX1      NEG   NUP<span class="m">-98</span>  p15<span class="o">/</span>p16
</span><span class="line">      <span class="m">10</span>       <span class="m">37</span>        <span class="m">5</span>       <span class="m">74</span>        <span class="m">1</span>        <span class="m">1</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Para seleccionar todos los genes de las células B con la fusión BCR/ABL y del grupo control</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">subset <span class="o">&lt;-</span> intersect<span class="p">(</span>grep<span class="p">(</span><span class="s">&quot;^B&quot;</span><span class="p">,</span> as.character<span class="p">(</span>ALL<span class="o">$</span>BT<span class="p">)),</span>
</span><span class="line">                    which<span class="p">(</span>as.character<span class="p">(</span>ALL<span class="o">$</span>mol.biol<span class="p">)</span>
</span><span class="line">                          <span class="o">%in%</span> c<span class="p">(</span><span class="s">&quot;BCR/ABL&quot;</span><span class="p">,</span> <span class="s">&quot;NEG&quot;</span><span class="p">)))</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Y creamos un nuevo set de expresión con los genes seleccionados previamente</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">eset <span class="o">&lt;-</span> ALL<span class="p">[,</span> subset<span class="p">]</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Para poder comparar el grupo de los pacientes control con los que presentan la fusión BCR/ABL, necesitamos convertir a factores la columna <code>eset$mol.biol</code></p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">eset<span class="o">$</span>mol.biol <span class="o">&lt;-</span> factor<span class="p">(</span>eset<span class="o">$</span>mol.biol<span class="p">)</span>
</span><span class="line">table<span class="p">(</span>eset<span class="o">$</span>mol.biol<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">BCR<span class="o">/</span>ABL     NEG
</span><span class="line">     <span class="m">37</span>      <span class="m">42</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Muchos de los genes en el chip no se expresan en las celulas B estudiados aquí, o podrían solo tener una pequeña variabilidad a traves de las sondas. Trataremos de remover esos genes con un filtro de intensidad (la intensidad de un gen debe de estar arriba de 100 y en al menos el 25 porciento de las sondas);  y un filtro de varianza (el rango intercuartil de las intensidades log2 debera ser al menos de 0.5)</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">library<span class="p">(</span>genefilter<span class="p">)</span>
</span><span class="line">f1 <span class="o">&lt;-</span> pOverA<span class="p">(</span><span class="m">0.25</span><span class="p">,</span> log2<span class="p">(</span><span class="m">100</span><span class="p">))</span>
</span><span class="line">f2 <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>x<span class="p">)</span> <span class="p">(</span>IQR<span class="p">(</span>x<span class="p">)</span> <span class="o">&gt;</span> <span class="m">0.5</span><span class="p">)</span>
</span><span class="line">ff <span class="o">&lt;-</span> filterfun<span class="p">(</span>f1<span class="p">,</span> f2<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>
<p>Cuantos genes podemos obtener despues del filtrado?</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">selected <span class="o">&lt;-</span> genefilter<span class="p">(</span>eset<span class="p">,</span> ff<span class="p">)</span>
</span><span class="line">sum<span class="p">(</span>selected<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><code>[1] 2391</code></p>

<p>Crearemos un nuevo ExpressionSet que contenga solo los genes que pasaron nuestro filtro.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">esetSub <span class="o">&lt;-</span> eset<span class="p">[</span>selected<span class="p">,</span> <span class="p">]</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Ahora estamos listos para analizar la expresion diferencial en los genes seleccionados entre las muetraas BRC/ABL y la normales citogenéticamente. Utilizaremos una prueba t pareada, para identicar los genes expresados de manera diferencial entre los dos grupos.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">library<span class="p">(</span>multtest<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>La funcion <code>mt.teststat</code> del paquete multtest nos permite calcular varios estadsticos de prueba comunmente usados para todas las las de una matriz de datos (chequen su pagina de ayuda). Primero, calcularemos los p-values nominales</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">library<span class="p">(</span>multtest<span class="p">)</span>
</span><span class="line">cl <span class="o">&lt;-</span> as.numeric<span class="p">(</span>esetSub<span class="o">$</span>mol.biol <span class="o">==</span> <span class="s">&quot;BCR/ABL&quot;</span><span class="p">)</span>
</span><span class="line">t <span class="o">&lt;-</span> mt.teststat<span class="p">(</span>
</span><span class="line">exprs<span class="p">(</span>esetSub<span class="p">),</span>
</span><span class="line">classlabel <span class="o">=</span> cl<span class="p">,</span>
</span><span class="line">test <span class="o">=</span> <span class="s">&quot;t.equalvar&quot;</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>
<p>La funcion <code>pt</code> nos da la distribucion t.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">pt <span class="o">&lt;-</span> <span class="m">2</span> <span class="o">*</span> pt<span class="p">(</span><span class="o">-</span>abs<span class="p">(</span>t<span class="p">),</span> df <span class="o">=</span> ncol<span class="p">(</span>exprs<span class="p">(</span>esetSub<span class="p">)))</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Podemos obtener una impresion de la cantidad de genes expresados de manera diferencial, observando en un histograma de la distribucion del valor de p.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">hist<span class="p">(</span>pt<span class="p">,</span> <span class="m">50</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><img src="/images/histogram-pt.png" /></p>

<p>La función <code>mt.rawp2adjp</code> de el paquete multtest contiene distintos procedimeintos de prueba (Checa la página de ayuda para esta función). Para obtener el ajuste de p-value en términos de FDR (False Discovery Rate) utilizaremos el método de  Benjamini y Hochberg. ¿Cuántos genes obtendremos si imponemos un FDR de 0.1?</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">pAdjusted <span class="o">&lt;-</span> mt.rawp2adjp<span class="p">(</span>pt<span class="p">,</span> proc <span class="o">=</span> c<span class="p">(</span><span class="s">&quot;BH&quot;</span><span class="p">))</span>
</span><span class="line">sum<span class="p">(</span>pAdjusted<span class="o">$</span>adjp<span class="p">[,</span> <span class="s">&quot;BH&quot;</span><span class="p">]</span> <span class="o">&lt;</span> <span class="m">0.1</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><code>[1] 171</code></p>

<p>También esta función retorna los p-values ordenados de menor a mayor. Para obtenerlos ordenados utilizaremos:</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">pBH <span class="o">&lt;-</span> pAdjusted<span class="o">$</span>adjp<span class="p">[</span>order<span class="p">(</span>pAdjusted<span class="o">$</span>index<span class="p">),</span> <span class="s">&quot;BH&quot;</span><span class="p">]</span>
</span><span class="line">names<span class="p">(</span>pBH<span class="p">)</span><span class="o">&lt;-</span>featureNames<span class="p">(</span>esetSub<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>
<p>Ahora, queremos conocer cuáles genes son los más significativos, por medio de el análisis de sus valores crudos (raw) y sus p-values ajustados por  distintos métodos. Los símbolos de los genes son obtenidos por medio del paquete de anotación <code>hgu95av2</code>.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">library<span class="p">(</span>annotate<span class="p">)</span>
</span><span class="line">library<span class="p">(</span>hgu95av2.db<span class="p">)</span>
</span><span class="line">diff <span class="o">&lt;-</span> pAdjusted<span class="o">$</span>index<span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">]</span>
</span><span class="line">genesymbolsDiff <span class="o">&lt;-</span> unlist<span class="p">(</span>mget<span class="p">(</span>featureNames<span class="p">(</span>esetSub<span class="p">)[</span>diff<span class="p">],</span> hgu95av2SYMBOL<span class="p">))</span>
</span><span class="line">genesymbolsDiff
</span></code></pre></td></tr></table></div></figure></notextile></div>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
</pre></td><td class="code"><pre><code class="r"><span class="line"> <span class="m">1636</span>_g_at   <span class="m">39730</span>_at    <span class="m">1635</span>_at   <span class="m">40202</span>_at   <span class="m">37027</span>_at <span class="m">39837</span>_s_at
</span><span class="line">    <span class="s">&quot;ABL1&quot;</span>     <span class="s">&quot;ABL1&quot;</span>     <span class="s">&quot;ABL1&quot;</span>     <span class="s">&quot;KLF9&quot;</span>    <span class="s">&quot;AHNAK&quot;</span>   <span class="s">&quot;ZNF467&quot;</span>
</span><span class="line"><span class="m">40480</span>_s_at   <span class="m">33774</span>_at   <span class="m">36591</span>_at   <span class="m">37014</span>_at
</span><span class="line">     <span class="s">&quot;FYN&quot;</span>    <span class="s">&quot;CASP8&quot;</span>   <span class="s">&quot;TUBA4A&quot;</span>      <span class="s">&quot;MX1&quot;</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>El top 3 de las sondas representan al gen ABL1, el cual es afectado, por la traslocación caracterizada por mas sondas  BCR/ABL. Ahora queremos ver si existen otro conjunto de sondas que representen a este gen, y ver éste ha sido seleccionado por nuestro filtro no específico.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">geneSymbols <span class="o">=</span> unlist<span class="p">(</span>mget<span class="p">(</span>featureNames<span class="p">(</span>ALL<span class="p">),</span> hgu95av2SYMBOL<span class="p">))</span>
</span><span class="line">ABL1probes <span class="o">&lt;-</span> which<span class="p">(</span>geneSymbols <span class="o">==</span> <span class="s">&quot;ABL1&quot;</span><span class="p">)</span>
</span><span class="line">selected<span class="p">[</span>ABL1probes<span class="p">]</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">  <span class="m">1635</span>_at <span class="m">1636</span>_g_at <span class="m">1656</span>_s_at <span class="m">2040</span>_s_at <span class="m">2041</span>_i_at  <span class="m">39730</span>_at
</span><span class="line">     <span class="kc">TRUE</span>      <span class="kc">TRUE</span>     <span class="kc">FALSE</span>     <span class="kc">FALSE</span>     <span class="kc">FALSE</span>      <span class="kc">TRUE</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Así que los otros conjuntos de sondas en el chip representando ABL1 han sido eliminados por el filtro, debido a baja intensidad o baja varianza. Ahora queremos saber si esto indica también expresión diferencial del gen <em>ALB1</em>.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">tABL1 <span class="o">&lt;-</span> mt.teststat<span class="p">(</span>exprs<span class="p">(</span>eset<span class="p">)[</span>ABL1probes<span class="p">,</span> <span class="p">],</span> classlabel <span class="o">=</span> cl<span class="p">,</span>
</span><span class="line">test <span class="o">=</span> <span class="s">&quot;t.equalvar&quot;</span><span class="p">)</span>
</span><span class="line">ptABL1 <span class="o">&lt;-</span> <span class="m">2</span> <span class="o">*</span> pt<span class="p">(</span><span class="o">-</span>abs<span class="p">(</span>tABL1<span class="p">),</span> df <span class="o">=</span> ncol<span class="p">(</span>exprs<span class="p">(</span>esetSub<span class="p">))</span> <span class="o">-</span> <span class="m">2</span><span class="p">)</span>
</span><span class="line">sort<span class="p">(</span>ptABL1<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line"><span class="p">[</span><span class="m">1</span><span class="p">]</span> <span class="m">3.762489e-14</span> <span class="m">4.791997e-13</span> <span class="m">2.445693e-10</span> <span class="m">5.486259e-02</span> <span class="m">5.842693e-01</span>
</span><span class="line"><span class="p">[</span><span class="m">6</span><span class="p">]</span> <span class="m">7.570959e-01</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Vemos ahora que solo tres de las seis sondas para <em>ABL1</em> muestran evidencia (de hecho, evidencia muy fuerte) para expresión diferencial. Seria interesante seguir investigando si el conjunto de sondas, en lo que concierne pe. a su localización en la secuencia del transcripto <em>ABL1</em> - en efecto la fusión de genes <em>BCR/ABL</em> resulta de la traslocación del gen normal <em>ABL1</em>. Utilizando la ontología de genes Muchos de los efectos debidos a la traslocación <em>BCR/ABL</em>, son mediados por la actividad de tirosina cinasa. Veamos ahora los conjuntos de sondas en el chip que pertenecen a el término GO de actividad proteína tirosina cinasa, la cuál tiene el identificador <em>GO:0004713</em>.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">gN <span class="o">&lt;-</span> featureNames<span class="p">(</span>esetSub<span class="p">)</span>
</span><span class="line">tykin <span class="o">&lt;-</span> unique<span class="p">(</span>unlist<span class="p">(</span>lookUp<span class="p">(</span><span class="s">&quot;GO:0004713&quot;</span><span class="p">,</span>
</span><span class="line"><span class="s">&quot;hgu95av2&quot;</span><span class="p">,</span> <span class="s">&quot;GO2ALLPROBES&quot;</span><span class="p">)))</span>
</span><span class="line">str<span class="p">(</span>tykin<span class="p">)</span>
</span><span class="line">sel <span class="o">&lt;-</span> <span class="p">(</span>gN <span class="o">%in%</span> tykin<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Ahora podermos checar si los genes entre las tirisina cinasas son expresados de manera diferencial, respecto a los otros genes. Utilizaremos una prueba exacta de Fisher, para tablas de contingencias, para revisar las proporciones de los genes expresados diferencialmente son significativas entre dos grupos de genes.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">tab <span class="o">&lt;-</span> table<span class="p">(</span>pt <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">,</span> sel<span class="p">,</span> dnn <span class="o">=</span> c<span class="p">(</span><span class="s">&quot;p &lt; 0.05&quot;</span><span class="p">,</span> <span class="s">&quot;tykin&quot;</span><span class="p">))</span>
</span><span class="line">print<span class="p">(</span>tab<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">    tykin
</span><span class="line">p <span class="o">&lt;</span> <span class="m">0.05</span> <span class="kc">FALSE</span> <span class="kc">TRUE</span>
</span><span class="line">   <span class="kc">FALSE</span>  <span class="m">1914</span>   <span class="m">26</span>
</span><span class="line">   <span class="kc">TRUE</span>    <span class="m">434</span>   <span class="m">17</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">fisher.test<span class="p">(</span>tab<span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">Fisher<span class="s">&#39;</span><span class="err">s Exact Test for Count Data</span>
</span><span class="line">
</span><span class="line">data<span class="o">:</span>  tab
</span><span class="line">p<span class="o">-</span>value <span class="o">=</span> <span class="m">0.001297</span>
</span><span class="line">alternative hypothesis<span class="o">:</span> true odds ratio is not equal to <span class="m">1</span>
</span><span class="line"><span class="m">95</span> percent confidence interval<span class="o">:</span>
</span><span class="line"> <span class="m">1.453631</span> <span class="m">5.573555</span>
</span><span class="line">sample estimates<span class="o">:</span>
</span><span class="line">odds ratio
</span><span class="line">  <span class="m">2.881814</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p><strong>Análisis para el enriquecimiento de conjunto de genes</strong>. Ahora queremos buscar el enriquecimiento en la expresión diferencial en todas las categorías GO, no solo para la actividad de tirosina cinasa. Utilizaremos el paquete TopGO para este fin.  Y necesitaremos definir una función que seleccione los genes expresados de manera diferencial.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">library<span class="p">(</span>topGO<span class="p">)</span> <span class="c1">#load the package</span>
</span><span class="line">topDiffGenes <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>allScore<span class="p">)</span> <span class="kr">return</span><span class="p">(</span>allScore <span class="o">&lt;</span> <span class="m">0.05</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Ahora estamos listos para hecer un nuevo objeto <em>topGO</em>. Buscaremos en las ontologías en el nivel de función molecular MF.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">GOdataMF <span class="o">&lt;-</span> new<span class="p">(</span><span class="s">&quot;topGOdata&quot;</span><span class="p">,</span> ontology <span class="o">=</span> <span class="s">&quot;MF&quot;</span><span class="p">,</span>
</span><span class="line">allGenes <span class="o">=</span> pBH<span class="p">,</span> geneSel <span class="o">=</span> topDiffGenes<span class="p">,</span> annot <span class="o">=</span> annFUN.db<span class="p">,</span>
</span><span class="line">affyLib <span class="o">=</span> <span class="s">&quot;hgu95av2.db&quot;</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>
<p>Ahora podremos correr una prueba de <em>fisher</em>, para todos los procesos biológicos o utilizar algunos otros métodos (Bioinformatics, 2006, 22(13):1600-1607)</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">resultFisher <span class="o">&lt;-</span> runTest<span class="p">(</span>GOdataMF<span class="p">,</span> algorithm <span class="o">=</span> <span class="s">&quot;classic&quot;</span><span class="p">,</span> statistic <span class="o">=</span> <span class="s">&quot;fisher&quot;</span><span class="p">)</span>
</span><span class="line">resultKS<span class="o">&lt;-</span> runTest<span class="p">(</span>GOdataMF<span class="p">,</span> algorithm <span class="o">=</span> <span class="s">&quot;classic&quot;</span><span class="p">,</span> statistic <span class="o">=</span> <span class="s">&quot;ks&quot;</span><span class="p">)</span>
</span><span class="line">resultFisher.elim<span class="o">&lt;-</span> runTest<span class="p">(</span>GOdataMF<span class="p">,</span> algorithm <span class="o">=</span> <span class="s">&quot;elim&quot;</span><span class="p">,</span> statistic <span class="o">=</span> <span class="s">&quot;fisher&quot;</span><span class="p">)</span>
</span><span class="line">resultKS.elim<span class="o">&lt;-</span> runTest<span class="p">(</span>GOdataMF<span class="p">,</span> algorithm <span class="o">=</span> <span class="s">&quot;elim&quot;</span><span class="p">,</span> statistic <span class="o">=</span> <span class="s">&quot;ks&quot;</span><span class="p">)</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Para mostrar los resultados</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">allRes <span class="o">&lt;-</span> GenTable<span class="p">(</span>GOdataMF<span class="p">,</span> classicFisher <span class="o">=</span> resultFisher<span class="p">,</span>
</span><span class="line">classicKS <span class="o">=</span> resultKS<span class="p">,</span>elimFisher <span class="o">=</span> resultFisher.elim<span class="p">,</span> elimKS <span class="o">=</span> resultKS.elim<span class="p">,</span>
</span><span class="line">orderBy <span class="o">=</span> <span class="s">&quot;elimKS&quot;</span><span class="p">,</span> ranksOf <span class="o">=</span> <span class="s">&quot;classicFisher&quot;</span><span class="p">,</span> topNodes <span class="o">=</span> <span class="m">20</span><span class="p">)</span>
</span><span class="line">allRes
</span></code></pre></td></tr></table></div></figure></notextile></div>
<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
<span class="line-number">20</span>
<span class="line-number">21</span>
<span class="line-number">22</span>
<span class="line-number">23</span>
<span class="line-number">24</span>
<span class="line-number">25</span>
<span class="line-number">26</span>
<span class="line-number">27</span>
<span class="line-number">28</span>
<span class="line-number">29</span>
<span class="line-number">30</span>
<span class="line-number">31</span>
<span class="line-number">32</span>
<span class="line-number">33</span>
<span class="line-number">34</span>
<span class="line-number">35</span>
<span class="line-number">36</span>
<span class="line-number">37</span>
<span class="line-number">38</span>
<span class="line-number">39</span>
<span class="line-number">40</span>
<span class="line-number">41</span>
<span class="line-number">42</span>
<span class="line-number">43</span>
<span class="line-number">44</span>
<span class="line-number">45</span>
<span class="line-number">46</span>
<span class="line-number">47</span>
<span class="line-number">48</span>
<span class="line-number">49</span>
<span class="line-number">50</span>
<span class="line-number">51</span>
<span class="line-number">52</span>
<span class="line-number">53</span>
<span class="line-number">54</span>
<span class="line-number">55</span>
<span class="line-number">56</span>
<span class="line-number">57</span>
<span class="line-number">58</span>
<span class="line-number">59</span>
<span class="line-number">60</span>
<span class="line-number">61</span>
<span class="line-number">62</span>
<span class="line-number">63</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">    GO.ID                                        Term Annotated
</span><span class="line"><span class="m">1</span>  GO<span class="o">:</span><span class="m">0005509</span>                         calcium ion binding        <span class="m">78</span>
</span><span class="line"><span class="m">2</span>  GO<span class="o">:</span><span class="m">0004714</span> transmembrane receptor protein tyrosine <span class="kc">...</span>         <span class="m">7</span>
</span><span class="line"><span class="m">3</span>  GO<span class="o">:</span><span class="m">0003785</span>                       actin monomer binding         <span class="m">3</span>
</span><span class="line"><span class="m">4</span>  GO<span class="o">:</span><span class="m">0004515</span> nicotinate<span class="o">-</span>nucleotide adenylyltransferas...         <span class="m">3</span>
</span><span class="line"><span class="m">5</span>  GO<span class="o">:</span><span class="m">0051019</span>    mitogen<span class="o">-</span>activated protein kinase binding        <span class="m">10</span>
</span><span class="line"><span class="m">6</span>  GO<span class="o">:</span><span class="m">0004871</span>                  signal transducer activity       <span class="m">188</span>
</span><span class="line"><span class="m">7</span>  GO<span class="o">:</span><span class="m">0042379</span>                  chemokine receptor binding        <span class="m">10</span>
</span><span class="line"><span class="m">8</span>  GO<span class="o">:</span><span class="m">0038023</span>                 signaling receptor activity        <span class="m">91</span>
</span><span class="line"><span class="m">9</span>  GO<span class="o">:</span><span class="m">0005516</span>                          calmodulin binding        <span class="m">25</span>
</span><span class="line"><span class="m">10</span> GO<span class="o">:</span><span class="m">0008201</span>                             heparin binding        <span class="m">16</span>
</span><span class="line"><span class="m">11</span> GO<span class="o">:</span><span class="m">0005088</span> Ras guanyl<span class="o">-</span>nucleotide exchange factor ac...        <span class="m">26</span>
</span><span class="line"><span class="m">12</span> GO<span class="o">:</span><span class="m">0015276</span>           ligand<span class="o">-</span>gated ion channel activity         <span class="m">9</span>
</span><span class="line"><span class="m">13</span> GO<span class="o">:</span><span class="m">0005520</span>          insulin<span class="o">-</span>like growth factor binding         <span class="m">4</span>
</span><span class="line"><span class="m">14</span> GO<span class="o">:</span><span class="m">0031730</span>             CCR5 chemokine receptor binding         <span class="m">8</span>
</span><span class="line"><span class="m">15</span> GO<span class="o">:</span><span class="m">0048020</span>              CCR chemokine receptor binding         <span class="m">8</span>
</span><span class="line"><span class="m">16</span> GO<span class="o">:</span><span class="m">0044325</span>                         ion channel binding        <span class="m">27</span>
</span><span class="line"><span class="m">17</span> GO<span class="o">:</span><span class="m">0030515</span>                              snoRNA binding         <span class="m">4</span>
</span><span class="line"><span class="m">18</span> GO<span class="o">:</span><span class="m">0003735</span>          structural constituent of ribosome        <span class="m">25</span>
</span><span class="line"><span class="m">19</span> GO<span class="o">:</span><span class="m">0050699</span>                           WW domain binding         <span class="m">5</span>
</span><span class="line"><span class="m">20</span> GO<span class="o">:</span><span class="m">0017112</span> Rab guanyl<span class="o">-</span>nucleotide exchange factor ac...         <span class="m">6</span>
</span><span class="line">   Significant Expected Rank <span class="kr">in</span> classicFisher classicFisher classicKS
</span><span class="line"><span class="m">1</span>            <span class="m">9</span>     <span class="m">3.57</span>                    <span class="m">20</span>       <span class="m">0.00802</span>   <span class="m">0.00017</span>
</span><span class="line"><span class="m">2</span>            <span class="m">1</span>     <span class="m">0.32</span>                   <span class="m">191</span>       <span class="m">0.27977</span>   <span class="m">0.00028</span>
</span><span class="line"><span class="m">3</span>            <span class="m">3</span>     <span class="m">0.14</span>                     <span class="m">3</span>       <span class="m">9.3e-05</span>   <span class="m">0.00250</span>
</span><span class="line"><span class="m">4</span>            <span class="m">3</span>     <span class="m">0.14</span>                     <span class="m">4</span>       <span class="m">9.3e-05</span>   <span class="m">0.00250</span>
</span><span class="line"><span class="m">5</span>            <span class="m">3</span>     <span class="m">0.46</span>                    <span class="m">21</span>       <span class="m">0.00882</span>   <span class="m">0.00354</span>
</span><span class="line"><span class="m">6</span>           <span class="m">12</span>     <span class="m">8.60</span>                   <span class="m">147</span>       <span class="m">0.14534</span>   <span class="m">4.2e-06</span>
</span><span class="line"><span class="m">7</span>            <span class="m">1</span>     <span class="m">0.46</span>                   <span class="m">235</span>       <span class="m">0.37447</span>   <span class="m">0.00364</span>
</span><span class="line"><span class="m">8</span>            <span class="m">9</span>     <span class="m">4.16</span>                    <span class="m">38</span>       <span class="m">0.02102</span>   <span class="m">0.00050</span>
</span><span class="line"><span class="m">9</span>            <span class="m">1</span>     <span class="m">1.14</span>                   <span class="m">320</span>       <span class="m">0.69179</span>   <span class="m">0.00747</span>
</span><span class="line"><span class="m">10</span>           <span class="m">1</span>     <span class="m">0.73</span>                   <span class="m">279</span>       <span class="m">0.52843</span>   <span class="m">0.00801</span>
</span><span class="line"><span class="m">11</span>           <span class="m">5</span>     <span class="m">1.19</span>                    <span class="m">15</span>       <span class="m">0.00554</span>   <span class="m">0.00919</span>
</span><span class="line"><span class="m">12</span>           <span class="m">0</span>     <span class="m">0.41</span>                   <span class="m">388</span>       <span class="m">1.00000</span>   <span class="m">0.00925</span>
</span><span class="line"><span class="m">13</span>           <span class="m">1</span>     <span class="m">0.18</span>                   <span class="m">152</span>       <span class="m">0.17089</span>   <span class="m">0.00944</span>
</span><span class="line"><span class="m">14</span>           <span class="m">1</span>     <span class="m">0.37</span>                   <span class="m">204</span>       <span class="m">0.31282</span>   <span class="m">0.01124</span>
</span><span class="line"><span class="m">15</span>           <span class="m">1</span>     <span class="m">0.37</span>                   <span class="m">205</span>       <span class="m">0.31282</span>   <span class="m">0.01124</span>
</span><span class="line"><span class="m">16</span>           <span class="m">7</span>     <span class="m">1.23</span>                     <span class="m">6</span>       <span class="m">0.00014</span>   <span class="m">0.01204</span>
</span><span class="line"><span class="m">17</span>           <span class="m">0</span>     <span class="m">0.18</span>                   <span class="m">389</span>       <span class="m">1.00000</span>   <span class="m">0.01243</span>
</span><span class="line"><span class="m">18</span>           <span class="m">0</span>     <span class="m">1.14</span>                   <span class="m">390</span>       <span class="m">1.00000</span>   <span class="m">0.01295</span>
</span><span class="line"><span class="m">19</span>           <span class="m">0</span>     <span class="m">0.23</span>                   <span class="m">391</span>       <span class="m">1.00000</span>   <span class="m">0.01346</span>
</span><span class="line"><span class="m">20</span>           <span class="m">2</span>     <span class="m">0.27</span>                    <span class="m">47</span>       <span class="m">0.02755</span>   <span class="m">0.01445</span>
</span><span class="line">   elimFisher  elimKS
</span><span class="line"><span class="m">1</span>     <span class="m">0.00802</span> <span class="m">0.00017</span>
</span><span class="line"><span class="m">2</span>     <span class="m">0.27977</span> <span class="m">0.00028</span>
</span><span class="line"><span class="m">3</span>     <span class="m">9.3e-05</span> <span class="m">0.00250</span>
</span><span class="line"><span class="m">4</span>     <span class="m">9.3e-05</span> <span class="m">0.00250</span>
</span><span class="line"><span class="m">5</span>     <span class="m">0.00882</span> <span class="m">0.00354</span>
</span><span class="line"><span class="m">6</span>     <span class="m">0.52856</span> <span class="m">0.00360</span>
</span><span class="line"><span class="m">7</span>     <span class="m">0.37447</span> <span class="m">0.00364</span>
</span><span class="line"><span class="m">8</span>     <span class="m">0.27691</span> <span class="m">0.00717</span>
</span><span class="line"><span class="m">9</span>     <span class="m">0.69179</span> <span class="m">0.00747</span>
</span><span class="line"><span class="m">10</span>    <span class="m">0.52843</span> <span class="m">0.00801</span>
</span><span class="line"><span class="m">11</span>    <span class="m">0.00554</span> <span class="m">0.00919</span>
</span><span class="line"><span class="m">12</span>    <span class="m">1.00000</span> <span class="m">0.00925</span>
</span><span class="line"><span class="m">13</span>    <span class="m">0.17089</span> <span class="m">0.00944</span>
</span><span class="line"><span class="m">14</span>    <span class="m">0.31282</span> <span class="m">0.01124</span>
</span><span class="line"><span class="m">15</span>    <span class="m">0.31282</span> <span class="m">0.01124</span>
</span><span class="line"><span class="m">16</span>    <span class="m">0.00014</span> <span class="m">0.01204</span>
</span><span class="line"><span class="m">17</span>    <span class="m">1.00000</span> <span class="m">0.01243</span>
</span><span class="line"><span class="m">18</span>    <span class="m">1.00000</span> <span class="m">0.01295</span>
</span><span class="line"><span class="m">19</span>    <span class="m">1.00000</span> <span class="m">0.01346</span>
</span><span class="line"><span class="m">20</span>    <span class="m">0.02755</span> <span class="m">0.01445</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Por último queremos buscar genes en una de las categorías GO. La función printGenes, está hecha para esa tarea. Hagámoslo para la actividad proteína tirosina cinasa (<em>GO:0004713</em>) como un ejemplo.</p>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">gt <span class="o">&lt;-</span> printGenes<span class="p">(</span>GOdataMF<span class="p">,</span>
</span><span class="line">whichTerms <span class="o">=</span> <span class="s">&quot;GO:0004713&quot;</span><span class="p">,</span> chip <span class="o">=</span> <span class="s">&quot;hgu95av2.db&quot;</span><span class="p">,</span>
</span><span class="line">numChar <span class="o">=</span> <span class="m">40</span><span class="p">)</span>
</span><span class="line">gt
</span></code></pre></td></tr></table></div></figure></notextile></div>

<div class="bogus-wrapper"><notextile><figure class="code"><figcaption><span></span></figcaption><div class="highlight"><table><tr><td class="gutter"><pre class="line-numbers"><span class="line-number">1</span>
<span class="line-number">2</span>
<span class="line-number">3</span>
<span class="line-number">4</span>
<span class="line-number">5</span>
<span class="line-number">6</span>
<span class="line-number">7</span>
<span class="line-number">8</span>
<span class="line-number">9</span>
<span class="line-number">10</span>
<span class="line-number">11</span>
<span class="line-number">12</span>
<span class="line-number">13</span>
<span class="line-number">14</span>
<span class="line-number">15</span>
<span class="line-number">16</span>
<span class="line-number">17</span>
<span class="line-number">18</span>
<span class="line-number">19</span>
<span class="line-number">20</span>
<span class="line-number">21</span>
<span class="line-number">22</span>
<span class="line-number">23</span>
<span class="line-number">24</span>
<span class="line-number">25</span>
<span class="line-number">26</span>
<span class="line-number">27</span>
<span class="line-number">28</span>
<span class="line-number">29</span>
<span class="line-number">30</span>
<span class="line-number">31</span>
<span class="line-number">32</span>
<span class="line-number">33</span>
<span class="line-number">34</span>
<span class="line-number">35</span>
<span class="line-number">36</span>
<span class="line-number">37</span>
<span class="line-number">38</span>
<span class="line-number">39</span>
<span class="line-number">40</span>
<span class="line-number">41</span>
<span class="line-number">42</span>
<span class="line-number">43</span>
<span class="line-number">44</span>
<span class="line-number">45</span>
<span class="line-number">46</span>
<span class="line-number">47</span>
<span class="line-number">48</span>
<span class="line-number">49</span>
<span class="line-number">50</span>
<span class="line-number">51</span>
<span class="line-number">52</span>
<span class="line-number">53</span>
<span class="line-number">54</span>
<span class="line-number">55</span>
<span class="line-number">56</span>
<span class="line-number">57</span>
<span class="line-number">58</span>
<span class="line-number">59</span>
<span class="line-number">60</span>
<span class="line-number">61</span>
<span class="line-number">62</span>
<span class="line-number">63</span>
<span class="line-number">64</span>
<span class="line-number">65</span>
<span class="line-number">66</span>
<span class="line-number">67</span>
<span class="line-number">68</span>
<span class="line-number">69</span>
<span class="line-number">70</span>
<span class="line-number">71</span>
<span class="line-number">72</span>
<span class="line-number">73</span>
<span class="line-number">74</span>
<span class="line-number">75</span>
<span class="line-number">76</span>
<span class="line-number">77</span>
<span class="line-number">78</span>
<span class="line-number">79</span>
<span class="line-number">80</span>
<span class="line-number">81</span>
<span class="line-number">82</span>
<span class="line-number">83</span>
<span class="line-number">84</span>
<span class="line-number">85</span>
<span class="line-number">86</span>
<span class="line-number">87</span>
<span class="line-number">88</span>
<span class="line-number">89</span>
<span class="line-number">90</span>
<span class="line-number">91</span>
<span class="line-number">92</span>
<span class="line-number">93</span>
<span class="line-number">94</span>
</pre></td><td class="code"><pre><code class="r"><span class="line">          Chip ID LL.id Symbol.id
</span><span class="line"><span class="m">1636</span>_g_at   <span class="m">1636</span>_g_at    <span class="m">25</span>      ABL1
</span><span class="line"><span class="m">39730</span>_at     <span class="m">39730</span>_at    <span class="m">25</span>      ABL1
</span><span class="line"><span class="m">1635</span>_at       <span class="m">1635</span>_at    <span class="m">25</span>      ABL1
</span><span class="line"><span class="m">40480</span>_s_at <span class="m">40480</span>_s_at  <span class="m">2534</span>       FYN
</span><span class="line"><span class="m">2039</span>_s_at   <span class="m">2039</span>_s_at  <span class="m">2534</span>       FYN
</span><span class="line"><span class="m">754</span>_s_at     <span class="m">754</span>_s_at   <span class="m">613</span>       BCR
</span><span class="line"><span class="m">36643</span>_at     <span class="m">36643</span>_at    <span class="kc">NA</span>      <span class="o">&lt;</span><span class="kc">NA</span><span class="o">&gt;</span>
</span><span class="line"><span class="m">38662</span>_at     <span class="m">38662</span>_at  <span class="m">5610</span>   EIF2AK2
</span><span class="line"><span class="m">34877</span>_at     <span class="m">34877</span>_at  <span class="m">3716</span>      JAK1
</span><span class="line"><span class="m">537</span>_f_at     <span class="m">537</span>_f_at   <span class="m">613</span>       BCR
</span><span class="line"><span class="m">2057</span>_g_at   <span class="m">2057</span>_g_at  <span class="m">2260</span>     FGFR1
</span><span class="line"><span class="m">1007</span>_s_at   <span class="m">1007</span>_s_at    <span class="kc">NA</span>      <span class="o">&lt;</span><span class="kc">NA</span><span class="o">&gt;</span>
</span><span class="line"><span class="m">760</span>_at         <span class="m">760</span>_at  <span class="m">8445</span>     DYRK2
</span><span class="line"><span class="m">1333</span>_f_at   <span class="m">1333</span>_f_at   <span class="m">613</span>       BCR
</span><span class="line"><span class="m">2056</span>_at       <span class="m">2056</span>_at  <span class="m">2260</span>     FGFR1
</span><span class="line"><span class="m">854</span>_at         <span class="m">854</span>_at   <span class="m">640</span>       BLK
</span><span class="line"><span class="m">41594</span>_at     <span class="m">41594</span>_at  <span class="m">3716</span>      JAK1
</span><span class="line"><span class="m">1065</span>_at       <span class="m">1065</span>_at  <span class="m">2322</span>      FLT3
</span><span class="line"><span class="m">34583</span>_at     <span class="m">34583</span>_at  <span class="m">2322</span>      FLT3
</span><span class="line"><span class="m">33804</span>_at     <span class="m">33804</span>_at  <span class="m">2185</span>     PTK2B
</span><span class="line"><span class="m">40742</span>_at     <span class="m">40742</span>_at  <span class="m">3055</span>       HCK
</span><span class="line"><span class="m">1498</span>_at       <span class="m">1498</span>_at  <span class="m">7535</span>     ZAP70
</span><span class="line"><span class="m">40936</span>_at     <span class="m">40936</span>_at <span class="m">51232</span>     CRIM1
</span><span class="line"><span class="m">36264</span>_at     <span class="m">36264</span>_at  <span class="m">4145</span>      MATK
</span><span class="line"><span class="m">1844</span>_s_at   <span class="m">1844</span>_s_at  <span class="m">5604</span>    MAP2K1
</span><span class="line"><span class="m">1810</span>_s_at   <span class="m">1810</span>_s_at  <span class="m">5580</span>     PRKCD
</span><span class="line"><span class="m">36909</span>_at     <span class="m">36909</span>_at  <span class="m">7465</span>      WEE1
</span><span class="line"><span class="m">993</span>_at         <span class="m">993</span>_at  <span class="m">7297</span>      TYK2
</span><span class="line"><span class="m">39931</span>_at     <span class="m">39931</span>_at  <span class="m">8444</span>     DYRK3
</span><span class="line"><span class="m">36115</span>_at     <span class="m">36115</span>_at  <span class="m">1198</span>      CLK3
</span><span class="line"><span class="m">2059</span>_s_at   <span class="m">2059</span>_s_at  <span class="m">3932</span>       LCK
</span><span class="line"><span class="m">2075</span>_s_at   <span class="m">2075</span>_s_at    <span class="kc">NA</span>      <span class="o">&lt;</span><span class="kc">NA</span><span class="o">&gt;</span>
</span><span class="line"><span class="m">33238</span>_at     <span class="m">33238</span>_at  <span class="m">3932</span>       LCK
</span><span class="line"><span class="m">292</span>_s_at     <span class="m">292</span>_s_at  <span class="m">1195</span>      CLK1
</span><span class="line"><span class="m">32833</span>_at     <span class="m">32833</span>_at  <span class="m">1195</span>      CLK1
</span><span class="line"><span class="m">1622</span>_at       <span class="m">1622</span>_at  <span class="m">5606</span>    MAP2K3
</span><span class="line"><span class="m">1768</span>_s_at   <span class="m">1768</span>_s_at  <span class="m">1445</span>       CSK
</span><span class="line"><span class="m">38118</span>_at     <span class="m">38118</span>_at  <span class="m">6464</span>      SHC1
</span><span class="line"><span class="m">1008</span>_f_at   <span class="m">1008</span>_f_at  <span class="m">5610</span>   EIF2AK2
</span><span class="line"><span class="m">1512</span>_at       <span class="m">1512</span>_at  <span class="m">1859</span>    DYRK1A
</span><span class="line"><span class="m">36946</span>_at     <span class="m">36946</span>_at  <span class="m">1859</span>    DYRK1A
</span><span class="line"><span class="m">1630</span>_s_at   <span class="m">1630</span>_s_at  <span class="m">6850</span>       SYK
</span><span class="line"><span class="m">1130</span>_at       <span class="m">1130</span>_at  <span class="m">5604</span>    MAP2K1
</span><span class="line"><span class="m">1402</span>_at       <span class="m">1402</span>_at  <span class="m">4067</span>       LYN
</span><span class="line"><span class="m">32616</span>_at     <span class="m">32616</span>_at  <span class="m">4067</span>       LYN
</span><span class="line"><span class="m">36885</span>_at     <span class="m">36885</span>_at  <span class="m">6850</span>       SYK
</span><span class="line">                                         Gene name raw p<span class="o">-</span>value
</span><span class="line"><span class="m">1636</span>_g_at  c<span class="o">-</span>abl oncogene <span class="m">1</span><span class="p">,</span> non<span class="o">-</span>receptor tyrosine <span class="kc">...</span>    <span class="m">9.00e-11</span>
</span><span class="line"><span class="m">39730</span>_at   c<span class="o">-</span>abl oncogene <span class="m">1</span><span class="p">,</span> non<span class="o">-</span>receptor tyrosine <span class="kc">...</span>    <span class="m">5.73e-10</span>
</span><span class="line"><span class="m">1635</span>_at    c<span class="o">-</span>abl oncogene <span class="m">1</span><span class="p">,</span> non<span class="o">-</span>receptor tyrosine <span class="kc">...</span>    <span class="m">1.95e-07</span>
</span><span class="line"><span class="m">40480</span>_s_at       FYN oncogene related to SRC<span class="p">,</span> FGR<span class="p">,</span> YES    <span class="m">0.000341</span>
</span><span class="line"><span class="m">2039</span>_s_at        FYN oncogene related to SRC<span class="p">,</span> FGR<span class="p">,</span> YES    <span class="m">0.002151</span>
</span><span class="line"><span class="m">754</span>_s_at                     breakpoint cluster region    <span class="m">0.027205</span>
</span><span class="line"><span class="m">36643</span>_at                                            <span class="kc">NA</span>    <span class="m">0.028651</span>
</span><span class="line"><span class="m">38662</span>_at   eukaryotic translation initiation factor...    <span class="m">0.036347</span>
</span><span class="line"><span class="m">34877</span>_at                                Janus kinase <span class="m">1</span>    <span class="m">0.048349</span>
</span><span class="line"><span class="m">537</span>_f_at                     breakpoint cluster region    <span class="m">0.065386</span>
</span><span class="line"><span class="m">2057</span>_g_at          fibroblast growth factor receptor <span class="m">1</span>    <span class="m">0.065447</span>
</span><span class="line"><span class="m">1007</span>_s_at                                           <span class="kc">NA</span>    <span class="m">0.080622</span>
</span><span class="line"><span class="m">760</span>_at     dual<span class="o">-</span>specificity tyrosine<span class="o">-</span><span class="p">(</span>Y<span class="p">)</span><span class="o">-</span>phosphoryl...    <span class="m">0.084581</span>
</span><span class="line"><span class="m">1333</span>_f_at                    breakpoint cluster region    <span class="m">0.116256</span>
</span><span class="line"><span class="m">2056</span>_at            fibroblast growth factor receptor <span class="m">1</span>    <span class="m">0.136657</span>
</span><span class="line"><span class="m">854</span>_at                      B lymphoid tyrosine kinase    <span class="m">0.142854</span>
</span><span class="line"><span class="m">41594</span>_at                                Janus kinase <span class="m">1</span>    <span class="m">0.158560</span>
</span><span class="line"><span class="m">1065</span>_at                  fms<span class="o">-</span>related tyrosine kinase <span class="m">3</span>    <span class="m">0.224691</span>
</span><span class="line"><span class="m">34583</span>_at                 fms<span class="o">-</span>related tyrosine kinase <span class="m">3</span>    <span class="m">0.225452</span>
</span><span class="line"><span class="m">33804</span>_at                protein tyrosine kinase <span class="m">2</span> beta    <span class="m">0.288513</span>
</span><span class="line"><span class="m">40742</span>_at                       hemopoietic cell kinase    <span class="m">0.328005</span>
</span><span class="line"><span class="m">1498</span>_at    zeta<span class="o">-</span>chain <span class="p">(</span>TCR<span class="p">)</span> associated protein kina...    <span class="m">0.339735</span>
</span><span class="line"><span class="m">40936</span>_at   cysteine rich transmembrane BMP regulato...    <span class="m">0.340416</span>
</span><span class="line"><span class="m">36264</span>_at      megakaryocyte<span class="o">-</span>associated tyrosine kinase    <span class="m">0.439155</span>
</span><span class="line"><span class="m">1844</span>_s_at  mitogen<span class="o">-</span>activated protein kinase kinase <span class="kc">...</span>    <span class="m">0.486525</span>
</span><span class="line"><span class="m">1810</span>_s_at                      protein kinase C<span class="p">,</span> delta    <span class="m">0.502015</span>
</span><span class="line"><span class="m">36909</span>_at                     WEE1 G2 checkpoint kinase    <span class="m">0.577787</span>
</span><span class="line"><span class="m">993</span>_at                               tyrosine kinase <span class="m">2</span>    <span class="m">0.577787</span>
</span><span class="line"><span class="m">39931</span>_at   dual<span class="o">-</span>specificity tyrosine<span class="o">-</span><span class="p">(</span>Y<span class="p">)</span><span class="o">-</span>phosphoryl...    <span class="m">0.659157</span>
</span><span class="line"><span class="m">36115</span>_at                             CDC<span class="o">-</span>like kinase <span class="m">3</span>    <span class="m">0.710543</span>
</span><span class="line"><span class="m">2059</span>_s_at  lymphocyte<span class="o">-</span>specific protein tyrosine kin...    <span class="m">0.727480</span>
</span><span class="line"><span class="m">2075</span>_s_at                                           <span class="kc">NA</span>    <span class="m">0.754511</span>
</span><span class="line"><span class="m">33238</span>_at   lymphocyte<span class="o">-</span>specific protein tyrosine kin...    <span class="m">0.768520</span>
</span><span class="line"><span class="m">292</span>_s_at                             CDC<span class="o">-</span>like kinase <span class="m">1</span>    <span class="m">0.786830</span>
</span><span class="line"><span class="m">32833</span>_at                             CDC<span class="o">-</span>like kinase <span class="m">1</span>    <span class="m">0.799613</span>
</span><span class="line"><span class="m">1622</span>_at    mitogen<span class="o">-</span>activated protein kinase kinase <span class="kc">...</span>    <span class="m">0.807841</span>
</span><span class="line"><span class="m">1768</span>_s_at                        c<span class="o">-</span>src tyrosine kinase    <span class="m">0.888941</span>
</span><span class="line"><span class="m">38118</span>_at   SHC <span class="p">(</span>Src homology <span class="m">2</span> domain containing<span class="p">)</span> t...    <span class="m">0.896863</span>
</span><span class="line"><span class="m">1008</span>_f_at  eukaryotic translation initiation factor...    <span class="m">0.909363</span>
</span><span class="line"><span class="m">1512</span>_at    dual<span class="o">-</span>specificity tyrosine<span class="o">-</span><span class="p">(</span>Y<span class="p">)</span><span class="o">-</span>phosphoryl...    <span class="m">0.918548</span>
</span><span class="line"><span class="m">36946</span>_at   dual<span class="o">-</span>specificity tyrosine<span class="o">-</span><span class="p">(</span>Y<span class="p">)</span><span class="o">-</span>phosphoryl...    <span class="m">0.924559</span>
</span><span class="line"><span class="m">1630</span>_s_at                       spleen tyrosine kinase    <span class="m">0.929048</span>
</span><span class="line"><span class="m">1130</span>_at    mitogen<span class="o">-</span>activated protein kinase kinase <span class="kc">...</span>    <span class="m">0.932251</span>
</span><span class="line"><span class="m">1402</span>_at    v<span class="o">-</span>yes<span class="m">-1</span> Yamaguchi sarcoma viral related <span class="kc">...</span>    <span class="m">0.969757</span>
</span><span class="line"><span class="m">32616</span>_at   v<span class="o">-</span>yes<span class="m">-1</span> Yamaguchi sarcoma viral related <span class="kc">...</span>    <span class="m">0.981146</span>
</span><span class="line"><span class="m">36885</span>_at                        spleen tyrosine kinase    <span class="m">0.999958</span>
</span></code></pre></td></tr></table></div></figure></notextile></div>

<p>Este ejercicio está basado y adaptado en una práctica de Stale Nygard de Bioinformatics core facility, OUS/UiO.</p>

<ol class="bibliography"><li><span id="Chiaretti2005">Chiaretti, S. (2005). Gene Expression Profiles of B-lineage Adult Acute Lymphocytic Leukemia Reveal Genetic Patterns that Identify Lineage Derivation and Distinct Mechanisms of Transformation. <i>Clinical Cancer Research</i>, <i>11</i>(20), 7209–7219. doi:10.1158/1078-0432.ccr-04-2165</span></li>
<li><span id="Nygaard2011">Nygraard, S. (2011). Microarray data lab. Retrieved from http://friveroll.github.io/pdfs/R-lab-sn.pdf</span></li></ol>

<p>Código Completo:</p>

<div><script src="https://gist.github.com/2691965.js"></script>
<noscript><pre><code>#Analisis de microarreglos

#Instalar programas requeridos
source(&quot;http://bioconductor.org/biocLite.R&quot;)
biocLite(c(&quot;affy&quot;,&quot;ALL&quot;,&quot;annotate&quot;,&quot;hgu95av2.db&quot;,&quot;multtest&quot;,&quot;topGO&quot;,&quot;genefilter&quot;))

#Cargar la libreria ALL
library(ALL)
data(ALL)
#Dimension del objeto
dim(exprs(ALL))

#Definir un objeto ExpressionSet con todas las muestras
table(ALL$BT)
table(ALL$mol.biol)
subset &lt;- intersect(grep(&quot;^B&quot;, as.character(ALL$BT)),
                    which(as.character(ALL$mol.biol) %in% c(&quot;BCR/ABL&quot;, &quot;NEG&quot;)))
eset &lt;- ALL[, subset]
eset$mol.biol &lt;- factor(eset$mol.biol)
table(eset$mol.biol)

#Filtro 

library(genefilter)
f1 &lt;- pOverA(0.25, log2(100))
f2 &lt;- function(x) (IQR(x) &gt; 0.5)
ff &lt;- filterfun(f1, f2)
selected &lt;- genefilter(eset, ff)
sum(selected)
esetSub &lt;- eset[selected, ]

library(multtest)
cl &lt;- as.numeric(esetSub$mol.biol == &quot;BCR/ABL&quot;)
t &lt;- mt.teststat(exprs(esetSub), classlabel = cl, test = &quot;t.equalvar&quot;)
pt &lt;- 2 * pt(-abs(t), df = ncol(exprs(esetSub)) - 2)
hist(pt, 50)

pAdjusted &lt;- mt.rawp2adjp(pt, proc = c(&quot;BH&quot;))
sum(pAdjusted$adjp[, &quot;BH&quot;] &lt; 0.1)

pBH &lt;- pAdjusted$adjp[order(pAdjusted$index), &quot;BH&quot;]
names(pBH)&lt;-featureNames(esetSub)

library(annotate)
library(hgu95av2.db)
diff &lt;- pAdjusted$index[1:10]
genesymbolsDiff &lt;- unlist(mget(featureNames(esetSub)[diff], hgu95av2SYMBOL))
genesymbolsDiff

geneSymbols = unlist(mget(featureNames(ALL), hgu95av2SYMBOL))
ABL1probes &lt;- which(geneSymbols == &quot;ABL1&quot;)
selected[ABL1probes]

tABL1 &lt;- mt.teststat(exprs(eset)[ABL1probes, ], classlabel = cl,
                     test = &quot;t.equalvar&quot;)
ptABL1 &lt;- 2 * pt(-abs(tABL1), df = ncol(exprs(esetSub)) - 2)
sort(ptABL1)

gN &lt;- featureNames(esetSub)
tykin &lt;- unique(unlist(lookUp(&quot;GO:0004713&quot;, &quot;hgu95av2&quot;, &quot;GO2ALLPROBES&quot;)))
str(tykin)
sel &lt;- (gN %in% tykin)

tab &lt;- table(pt &lt; 0.05, sel, dnn = c(&quot;p &lt; 0.05&quot;, &quot;tykin&quot;))
print(tab)
fisher.test(tab)

library(topGO) #load the package

topDiffGenes &lt;- function(allScore) return(allScore &lt; 0.05)

GOdataMF &lt;- new(&quot;topGOdata&quot;, ontology = &quot;MF&quot;,
                allGenes = pBH, geneSel = topDiffGenes, annot = annFUN.db,
                affyLib = &quot;hgu95av2.db&quot;)

resultFisher &lt;- runTest(GOdataMF, algorithm = &quot;classic&quot;, statistic = &quot;fisher&quot;)

resultKS&lt;- runTest(GOdataMF, algorithm = &quot;classic&quot;, statistic = &quot;ks&quot;)
resultFisher.elim&lt;- runTest(GOdataMF, algorithm = &quot;elim&quot;, statistic = &quot;fisher&quot;)
resultKS.elim&lt;- runTest(GOdataMF, algorithm = &quot;elim&quot;, statistic = &quot;ks&quot;)

allRes &lt;- GenTable(GOdataMF, classicFisher = resultFisher,
                   classicKS = resultKS,elimFisher = resultFisher.elim, elimKS = resultKS.elim,
                   orderBy = &quot;elimKS&quot;, ranksOf = &quot;classicFisher&quot;, topNodes = 20)
allRes

gt &lt;- printGenes(GOdataMF, whichTerms = &quot;GO:0004713&quot;, chip = &quot;hgu95av2.db&quot;,
                 numChar = 40)
gt</code></pre></noscript></div>
</div>


  <footer>
    <p class="meta">
      
  

<span class="byline author vcard">Posted by <span class="fn">Felipe Antonio Riveroll Aguirre</span></span>

      








  


<time datetime="2012-05-20T21:16:00-05:00" pubdate data-updated="true">May 20<span>th</span>, 2012</time>
      

<span class="categories">
  
    <a class='category' href='/blog/categories/r/'>R</a>, <a class='category' href='/blog/categories/bioconductor/'>bioconductor</a>, <a class='category' href='/blog/categories/bioinformatica/'>bioinformática</a>, <a class='category' href='/blog/categories/microarreglos/'>microarreglos</a>
  
</span>


    </p>
    
      <div class="sharing">
  
  <a href="http://twitter.com/share" class="twitter-share-button" data-url="http://friveroll.github.io/blog/2012/05/20/introduccion-al-analisis-de-microarreglos/" data-via="FelipeRiveroll" data-counturl="http://friveroll.github.io/blog/2012/05/20/introduccion-al-analisis-de-microarreglos/" >Tweet</a>
  
  
  <div class="g-plusone" data-size="medium"></div>
  
  
    <div class="fb-like" data-send="true" data-width="450" data-show-faces="false"></div>
  
</div>

    
    <p class="meta">
      
        <a class="basic-alignment left" href="/blog/2011/07/24/comparacion-estructural-de-proteinas/" title="Previous Post: Comparación estructural de proteínas">&laquo; Comparación estructural de proteínas</a>
      
      
        <a class="basic-alignment right" href="/blog/2012/05/23/utilizando-r-para-analizar-datos-de-cinetica/" title="Next Post: Utilizando R para analizar datos de cinética">Utilizando R para analizar datos de cinética &raquo;</a>
      
    </p>
  </footer>
</article>

  <section>
    <h1>Comments</h1>
    <div id="disqus_thread" aria-live="polite"><noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>
  </section>

</div>

<aside class="sidebar">
  
    
<section>
	<span>
		<img src="http://www.gravatar.com/avatar/2e15ce001d0d215a6ce6f67fbb91eed2" alt="Gravatar of Felipe Antonio Riveroll Aguirre " title="Gravatar of Felipe Antonio Riveroll Aguirre" />
	</span>
</section>
<section>
  <h1>Recent Posts</h1>
  <ul id="recent_posts">
    
      <li class="post">
        <a href="/blog/2016/06/06/western-blotting-parte-ii-blotting/">Western Blotting Parte II: Blotting</a>
      </li>
    
      <li class="post">
        <a href="/blog/2015/03/02/western-blotting-parte-i-electroforesis/">Western Blotting Parte I: Electroforesis</a>
      </li>
    
      <li class="post">
        <a href="/blog/2014/06/23/modelos-animales-de-estres-en-farmacologia/">Modelos animales de estrés en farmacología</a>
      </li>
    
      <li class="post">
        <a href="/blog/2014/01/06/fisica-con-analisis-de-video-utilizando-logger-pro/">Física con Análisis de Video Utilizando Logger Pro</a>
      </li>
    
      <li class="post">
        <a href="/blog/2012/05/23/utilizando-r-para-analizar-datos-de-cinetica/">Utilizando R para analizar datos de cinética</a>
      </li>
    
  </ul>
</section>

<section>
  <h1>GitHub Repos</h1>
  <ul id="gh_repos">
    <li class="loading">Status updating...</li>
  </ul>
  
  <a href="https://github.com/friveroll">@friveroll</a> on GitHub
  
  <script type="text/javascript">
    $.domReady(function(){
        if (!window.jXHR){
            var jxhr = document.createElement('script');
            jxhr.type = 'text/javascript';
            jxhr.src = '/javascripts/libs/jXHR.js';
            var s = document.getElementsByTagName('script')[0];
            s.parentNode.insertBefore(jxhr, s);
        }

        github.showRepos({
            user: 'friveroll',
            count: 0,
            skip_forks: true,
            target: '#gh_repos'
        });
    });
  </script>
  <script src="/javascripts/github.js" type="text/javascript"> </script>
</section>


<section>
  <h1>On Delicious</h1>
  <div id="delicious"></div>
  <script type="text/javascript" src="http://feeds.delicious.com/v2/json/friveroll?count=3&amp;sort=date&amp;callback=renderDeliciousLinks"></script>
  <p><a href="http://delicious.com/friveroll">My Delicious Bookmarks &raquo;</a></p>
</section>


<section class="googleplus">
  <h1>
    <a href="https://plus.google.com/+FelipeRiverollAguirre?rel=author">
      <img src="http://www.google.com/images/icons/ui/gprofile_button-32.png" width="32" height="32">
      Google+
    </a>
  </h1>
</section>



<section class="twitterOct">
  <h1 style="margin-bottom: 0.4em"> Tweets </h1>
  <a class="twitter-timeline"
   data-dnt="true" href="https://twitter.com/FelipeRiveroll" 
     data-widget-id="423250617667051521">
     
     Tweets by @FelipeRiveroll
  </a>
  <script>!function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0],p=/^http:/.test(d.location)?'http':'https';if(!d.getElementById(id)){js=d.createElement(s);js.id=id;js.src=p+"://platform.twitter.com/widgets.js";fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");</script>
</section>

  
</aside>


    </div>
  </div>
  <footer role="contentinfo"><p>
  Copyright &copy; 2016 - Felipe Antonio Riveroll Aguirre -
  <span class="credit">Powered by <a href="http://octopress.org">Octopress</a></span>
</p>

</footer>
  

<script type="text/javascript">
      var disqus_shortname = 'friveroll';
      
        
        // var disqus_developer = 1;
        var disqus_identifier = 'http://friveroll.github.io/blog/2012/05/20/introduccion-al-analisis-de-microarreglos/';
        var disqus_url = 'http://friveroll.github.io/blog/2012/05/20/introduccion-al-analisis-de-microarreglos/';
        var disqus_script = 'embed.js';
      
    (function () {
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = 'http://' + disqus_shortname + '.disqus.com/' + disqus_script;
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    }());
</script>



<div id="fb-root"></div>
<script>(function(d, s, id) {
  var js, fjs = d.getElementsByTagName(s)[0];
  if (d.getElementById(id)) {return;}
  js = d.createElement(s); js.id = id;
  js.src = "//connect.facebook.net/en_US/all.js#appId=212934732101925&xfbml=1";
  fjs.parentNode.insertBefore(js, fjs);
}(document, 'script', 'facebook-jssdk'));</script>



  <script type="text/javascript">
    (function() {
      var script = document.createElement('script'); script.type = 'text/javascript'; script.async = true;
      script.src = 'https://apis.google.com/js/plusone.js';
      var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(script, s);
    })();
  </script>



  <script type="text/javascript">
    (function(){
      var twitterWidgets = document.createElement('script');
      twitterWidgets.type = 'text/javascript';
      twitterWidgets.async = true;
      twitterWidgets.src = 'http://platform.twitter.com/widgets.js';
      document.getElementsByTagName('head')[0].appendChild(twitterWidgets);
    })();
  </script>





</body>
</html>
